confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/event/ConfluencePackageReadEvent.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.confluence.filter.event;
+
+import org.xwiki.observation.event.AbstractCancelableEvent;
+
+/**
+ * Event emitted when the confluence package has been read.
+ *
+ * <ul>
+ * <li>source: the input filter stream that triggers the event.</li>
+ * <li>data: the confluence package that has been read.</li>
+ * </ul>
+ *
+ * @version $Id$
+ * @since 9.20.2
+ */
+public class ConfluencePackageReadEvent extends AbstractCancelableEvent
```

Comment:
It's a bit strange to have a cancellable event related to something that just finished (and so cannot be cancelled). Usually the cancellable event is mentioning what you are going to cancel (like an event about starting to produce filtering events, for example).

Reply:
A very good point. What about `ConfluencePackageFilteringEvent`.

Reply:
`ConfluenceFilteringEvent` is probably enough. Now, to be fully clean, having a `ConfluenceFilteringEvent` also calls for sending a corresponding `ConfluenceFilteredEvent`.